internal/metrics/metrics.go

```diff
@@ -0,0 +1,100 @@
+package metrics
+
+
+import (
+	"time"
+	"strconv"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const (
+	ms = 0.001
+	s  = 1
+	m  = 60 * s
+	kb = 1024
+	mb = 1024*kb
+)
+
+type metrics struct {
+	registry *prometheus.Registry
+
+	// Custom metics
+	request_durations *prometheus.HistogramVec
+	response_sizes    *prometheus.HistogramVec
+
+}
+
+/** Create a new metric instance
+ *
+ * The metrics are applied to a new non-global prometheus registry
+ */
+func NewMetrics() *metrics {
+	registry := prometheus.NewRegistry()
+	metrics := &metrics{
+		registry: registry,
+
+		request_durations: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "vdsslice_durations_histogram_seconds",
+			Help:    "VDSslice latency distributions.",
+			Buckets: []float64{100*ms, 500*ms, 1*s, 2*s, 5*s, 20*s, 1*m, 2*m},
+		}, []string{"method", "path", "status", "cachehit"}),
+
+		response_sizes: prometheus.NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "vdsslice_response_sizes_histogram_bytes",
+			Help:    "VDSslice response size distributions.",
+			Buckets: []float64{100*kb, 1*mb, 5*mb, 10*mb, 20*mb, 50*mb, 100*mb, 200*mb},
```

Comment:
Any idea why grafana shows us this
![image](https://user-images.githubusercontent.com/44965282/217539948-b78ba3e0-1128-4816-a96f-df6d6bf8f8fb.png)
when we have way bigger buckets. It shouldn't be able to distinguish between .
Are we wrong in grafana, or here?

I think something is with grafana as when you remove "quantile" you get different results in kilobytes.
But would be nice to confirm.